### PR TITLE
Added JSS

### PR DESCRIPTION
**Pull request by seanoch**

### CODE REVIEW

File: 3-JSS/scripts/style_jss.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+const lightGreen = 'rgb(150, 202, 167)';
+const darkRed = 'rgb(78, 25, 23)';
+const lightRed = 'rgb(204, 124, 121)';
+const darkYellow = 'rgb(104, 89, 2)';
+const lightYellow = 'rgb(194, 173, 53)';
+const darkGray = '#505050';
+const lightGray = '#b3b3b3';
+
+// One time setup with default plugins and settings.
+jss.setup(preset())
```
Comment (tombenezra) [inline 3-JSS/scripts/style_jss.js:19]: I suggest separating the setup and style definitions into different files

File: 3-JSS/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const CopyPlugin = require("copy-webpack-plugin");
+
+module.exports = {
+    mode: 'development',
```
Comment (tombenezra) [inline 3-JSS/webpack.config.js:6]: usually you would go with a production mode by default for bundling and development settings when working with the dev-server. can you tell the difference?

File: 3-JSS/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        clean: true
+    },
+    devtool: 'eval-source-map',
+    devServer: {
+        static: './dist',
+        port: 5001,
+        open: true,
+        hot: true
+    },
+    //loaders
```
Comment (tombenezra) [inline 3-JSS/webpack.config.js:22]: remove comments

File: 3-JSS/package.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  "description": "",
+  "scripts": {
+    "build": "webpack",
+    "start": "webpack serve"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "css-loader": "^6.6.0",
```
Comment (tombenezra) [inline 3-JSS/package.json:13]: remove unused dependencies

File: 3-JSS/package.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    "html-webpack-plugin": "^5.5.0",
+    "style-loader": "^3.3.1",
+    "webpack": "^5.68.0",
+    "webpack-cli": "^4.9.2",
+    "webpack-dev-server": "^4.7.4"
+  },
+  "dependencies": {
+    "color": "^4.2.1",
+    "copy-webpack-plugin": "^10.2.4",
```
Comment (tombenezra) [inline 3-JSS/package.json:22]: it makes sense to define a webpack plugin `dev dependency`. https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file

File: 3-JSS/scripts/localstorage_manager.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+function setMapToLocalStorage(map) {
+  localStorage.setItem(
+    LOCAL_STORAGE_KEY,
+    JSON.stringify(Array.from(map.entries()))
+  );
+}
+
+function getMapFromLocalStorage() {
+  let jsonObject = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY));
```
Comment (tombenezra) [inline 3-JSS/scripts/localstorage_manager.js:11]: `const`

File: 3-JSS/scripts/localstorage_manager.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  let map = jsonObject != null ? new Map(jsonObject) : new Map();
```
Comment (tombenezra) [inline 3-JSS/scripts/localstorage_manager.js:12]: consider using a regular json object so you won't have to do this "complex" serialize/deserialize every time

File: 3-JSS/scripts/localstorage_manager.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  setMapToLocalStorage(map);
+}
+
+function removeItem(id) {
+  let map = getMapFromLocalStorage();
+  map.delete(id);
+  setMapToLocalStorage(map);
+}
+
+function getItems() {
```
Comment (tombenezra) [inline 3-JSS/scripts/localstorage_manager.js:29]: what's the point of re-exporting it?

File: 3-JSS/scripts/todo_list.js [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+
+  function removeFromLocalStorage(id) {
+    if (canUseLocalStorage()) {
+      removeItem(id);
+    }
+  }
+
+  function loadLocalStorage() {
+    if (canUseLocalStorage()) {
+      let numericIds = [];
```
Comment (tombenezra) [inline 3-JSS/scripts/todo_list.js:230]: unused

File: 3-JSS/scripts/todo_list.js [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+
+      addOrUpdateItem(id, {
+        text: label.innerText,
+        check: isChecked,
+      });
+    }
+  }
+
+  function removeFromLocalStorage(id) {
+    if (canUseLocalStorage()) {
```
Comment (tombenezra) [inline 3-JSS/scripts/todo_list.js:223]: why not having this check inside the local storage actions themselves?

**Review by tombenezra — CHANGES_REQUESTED** [review]

File: 3-JSS/scripts/style_jss.js [hunk 11]
```diff
@@ -0,0 +1,60 @@
+import { jss } from './preset_jss'
```
Comment (tombenezra) [inline 3-JSS/scripts/style_jss.js:1]:
```suggestion
import jss from 'jss'
```

**Review by tombenezra — CHANGES_REQUESTED** [review]

File: 3-JSS/scripts/localstorage_manager.js (hunk 8 above)
Comment (seanoch) [reply 3-JSS/scripts/localstorage_manager.js:37]: It's now updated to deal with unavailable local storage

**Review by tombenezra — APPROVED** [review]

**Pull request merged**